tests/test_rules.py

```diff
@@ -158,6 +158,15 @@ def test_complete_unrule_offers_all_rules(hlwm, rules_count):
 
     assert call.stdout == '\n'.join(rules + ['-F', '--all']) + '\n'
 
+def test_complete_rule(hlwm):
+    global string_props, numeric_props, consequences
+    assert hlwm.complete('rule', partial=True) == sorted(
+        [ i + ' ' for i in '! not prepend once printlabel'.split(' ')]
```

Comment:
There's not supposed to be whitespace after opening brackets (another PEP-8 rule)